File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Promot user to give a height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Print the pyramids line by line
    for (int i = 0; i < height; i++)
    {
        // Print the initial space
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }

        // Print the left bricks
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        // Print the gap
        printf("  ");

        // Print the right bricks
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        printf("\n");
    }

}

File: pset6/mario/more/mario.py
from cs50 import get_int

height = 0
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break


def print_block(h):
    if h == 0:
        return

    print_block(h - 1)

    for i in range(height - h):
        print(" ", end="")
    for i in range(h):
        print("#", end="")

    print("  ", end="")

    for i in range(h):
        print("#", end="")

    print()


print_block(height)